### ISSUE TITLE
Should the label -1 be set to a array of -1?

### ISSUE THREAD

**Issue by zhujiagang** [facebookresearch/video-long-term-feature-banks#7]
Sorry to bother you again.
In your paper, you say a box may have no label assigned (-1), as is also provided in your  `ava_train_predicted_boxes.csv`. But I notice that in the code
https://github.com/facebookresearch/video-long-term-feature-banks/blob/623972f5c2ab85fb37496a62b5888466895c6559/lib/datasets/ava_data_input.py#L80
a box with label=-1 will finally get a label array with all zeros for the binary cross entropy loss. Is this may make the training harder? Can we make a array of all -1 in caffe2 for the binary cross entropy loss?

**Comment by chaoyuaw** [comment]
Hi @zhujiagang , I'm not sure if I understand your questions correctly. For AVA, we implement a multi-label binary classification problem. Each class has label either 0 (action not happening) or 1 (action happening). "-1" means no action assigned, so the labels for all classes are 0. Could you please explain why this might make training harder, and what the motivation is with setting it to "-1"? Thank you.

**Comment by zhujiagang** [comment]
Thanks for your reply.
 I found the boxes with label "-1" and person confidence >0.9 are about 3/100 of the predicted train boxes.

I think when we classify a person with a bounding box into actions, it must have at least an action label being positive, "1". So when a detected box without label assigned from gt boxes, I think it is due to our insufficient annotation efforts. It should be assigned at least an action label, but now, we don't know its labels ("0" or "1"). So we can just ignore it during training. By setting the labels to "-1" (a signal to ignore), the boxes without labels assigned will not contribute to the gradients. Setting all classes to "0" may make the training harder when the box does belong to an action.

And my idea may be incorrect, because the unlabeled boxes don't belong to any of the action classes of AVA. 

Meanwhile, the IoU threshold for assigning box labels can also affect the training, because 0.9 is a relatively high threshold. When a box has a smaller IoU (0.89) with gt, it will have no positive labels. I think it is a big jump of action labels, which may make the training difficult. By ignore all other boxes with -1, it just has less samples for model training, than discriminating between ambiguous class boundaries.

I am currently doing the experiments for validating my ideas, waiting for my news tomorrow.

**Comment by zhujiagang** [comment]
The performance of the model after all samples with label -1 being removed from the `predicted.csv` decreases by 0.5% on the validation set of ava2.2. Perhaps my idea does not hold.

**Comment by chaoyuaw** [comment]
Hi @zhujiagang , Many thanks for sharing the results! I think a possible hypothesis/explanation is that our detector isn't perfect and could have (possibly systematic) false positives. If we don't include such cases (and use GT boxes only) during training, at test time the model might not know what to do with false positives. Including false positives in training gives the model a chance to learn how to model those cases.
